src/servers/trigger.ts

```diff
@@ -178,16 +180,16 @@ export const onchainBlockEventHandler = async (height: number) => {
   logger.info(`finish block ${height} handler with ${txNumber} transactions`)
 }
 
-export const invoiceUpdateEventHandler = async (invoice: GetInvoiceResult) => {
+export const invoiceUpdateEventHandler = async (
+  invoice: SubscribeToInvoiceInvoiceUpdatedEvent,
+): Promise<boolean | ApplicationError> => {
   logger.info({ invoice }, "invoiceUpdateEventHandler")
-
-  if (!invoice.is_held) {
-    return
-  }
-
-  const paymentHash = invoice.id as PaymentHash
-
-  await Wallets.updatePendingInvoiceByPaymentHash({ paymentHash, logger })
+  return invoice.is_held
+    ? Wallets.updatePendingInvoiceByPaymentHash({
+        paymentHash: invoice.id as PaymentHash,
+        logger,
+      })
+    : false
```

Comment:
what is the scenario for which `invoice.is_held` will be false?

Reply:
I guess the first call of `listenerHoldInvoice`?

Reply:
Yup when the invoice is first created and when it is subsequently settled